### PR TITLE
jewel: RGW: Multipart upload may double the quota

### PR DESCRIPTION
**Pull request by mattbenjamin**
http://tracker.ceph.com/issues/21791

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@mdw-at-linuxbox trial backport, please check the size param

File: src/rgw/rgw_rados.h [hunk 1]
```diff
@@ -2330,10 +2330,11 @@ class RGWRados
         ceph::real_time delete_at;
         bool canceled;
         const string *user_data;
+        bool completeMultipart;
 
         MetaParams() : mtime(NULL), rmattrs(NULL), data(NULL), manifest(NULL), ptag(NULL),
                  remove_objs(NULL), category(RGW_OBJ_CATEGORY_MAIN), flags(0),
-                 if_match(NULL), if_nomatch(NULL), olh_epoch(0), canceled(false), user_data(nullptr) {}
+		       if_match(NULL), if_nomatch(NULL), olh_epoch(0), canceled(false), user_data(nullptr), completeMultipart(false) {}
```
Comment (amitkumar50) [inline src/rgw/rgw_rados.h]: tab

**Comment by smithfarm** [comment]
jenkins test this please

**Comment by smithfarm** [comment]
jenkins test this please (failures in test/osd/osd-recovery-scrub.sh and test/test_pidfile.sh)

**Comment by smithfarm** [comment]
Integration testing: http://tracker.ceph.com/issues/21742#note-11

Will post a summary to the PR when the testing is complete.

**Comment by smithfarm** [comment]
This PR passed two RGW runs at http://tracker.ceph.com/issues/21742#note-11

It will be merged soon, unless there any objections.

**Pull request merged**